utils/frame/benchmarking-cli/src/pallet/command.rs

```diff
@@ -508,20 +509,31 @@ impl PalletCmd {
 			self.print_summary(&batches, &storage_info, pov_modes.clone())
 		}
 
-		// Create the weights.rs file.
+		// Write all weight files and possibly error afterwards.
+		let mut soft_errors = Vec::new();
```

Comment:
This makes no sense? There can only be one error here. You can just directly return on error?

Reply:
The `write_results` is `Result<Result<` and I return directly on an outer error but keep the inner one as soft error.